### PR TITLE
Shell with config based machine

### PR DESCRIPTION
**Pull request by tuler**
This starts the configured machine with a shell

### CODE REVIEW

File: apps/cli/src/commands/shell.ts [hunk 1]
```diff
@@ -18,62 +18,64 @@ export default class Shell extends BaseCommand<typeof Shell> {
...
+            cmd: [],
+            entrypoint: ["/bin/bash"],
+            env: [],
+            workdir: "/",
+        };
+
+        // start with interactive mode on
+        config.machine.interactive = true;
+
+        /* why this?
```
Comment (endersonmaia) [inline apps/cli/src/commands/shell.ts:61]: `-it` is the enhanced mode and requires the `stty` binary

```
 -i or --htif-console-getchar
    run in interactive mode using a HTIF console device.

    NON REPRODUCIBLE OPTION, DON'T USE THIS OPTION IN PRODUCTION
(...)
  -it
    run in enhanced interactive mode using a VirtIO console device.
    the console is resizable, more responsive, and support more features
    than the -i option.

    like --virtio-console, but automatically appends init commands to forward
    TERM and LANG environment variables from the host to the guest,
    allowing the use of true colors and special characters (when supported).
```

File: apps/cli/src/machine.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        args.push("--assert-rolling-template");
+    }
+    if (finalHash) {
+        args.push("--final-hash");
+    }
+    if (info?.workdir) {
+        args.push(`--append-init=WORKDIR="${info.workdir}"`);
+    }
+    if (interactive) {
+        args.push("-it");
```
Comment (endersonmaia) [inline apps/cli/src/machine.ts:83]: we should check for `stty` like we did before

Comment (tuler) [reply apps/cli/src/machine.ts:83]: Is `stty` not installed in any platform?

Comment (endersonmaia) [reply apps/cli/src/machine.ts:83]: > What happens if this stty is not installed?

I don't know. And I'm not finding the reference I used when I implemented this.

Comment (tuler) [reply apps/cli/src/machine.ts:83]: Did you need to install `stty`?

Comment (endersonmaia) [reply apps/cli/src/machine.ts:83]: > Did you need to install stty?

No

File: apps/cli/src/commands/shell.ts [hunk 3]
```diff
@@ -18,62 +18,66 @@ export default class Shell extends BaseCommand<typeof Shell> {
...
+                throw new Error(
+                    `drive '${name}' not built, run '${this.config.bin} build'`,
+                );
+            }
         }
 
+        // create shell entrypoint
+        const info: ImageInfo = {
+            cmd: [],
+            entrypoint: ["/bin/bash"],
```
Comment (endersonmaia) [inline apps/cli/src/commands/shell.ts:55]:
```suggestion
            entrypoint: ["/bin/sh"],
```

**Review by endersonmaia — CHANGES_REQUESTED** [review]

File: apps/cli/src/commands/shell.ts [hunk 4]
```diff
@@ -24,8 +24,14 @@ export default class Shell extends BaseCommand<typeof Shell> {
...
         "run-as-root": Flags.boolean({
-            description: "run as root user",
             default: false,
+            description: "run as root user",
+            summary: "run the cartesi machine as the root user",
+        }),
+        shell: Flags.string({
+            default: "/bin/sh",
+            description: "shell command to run",
+            summary: "shell to run",
```
Comment (endersonmaia) [inline apps/cli/src/commands/shell.ts:34]:
```suggestion
        command: Flags.string({
            default: "/bin/sh",
            description: "command to run",
            summary: "command to run",
            char: "C",
```

Won't it be strang `cartesi shell --shell=/bin/bash`  ?

I think that a `--command` would be better, and it's like `flyctl` does with `flyctl ssh console -C | --command`.

**Review by endersonmaia — CHANGES_REQUESTED** [review]

**Review by endersonmaia — APPROVED** [review]

**Pull request merged**